numpy/ma/core.py

```diff
@@ -3829,10 +3833,10 @@ def compress(self, condition, axis=None, out=None):
             _new._mask = _mask.compress(condition, axis=axis)
         return _new
 
-    def __str__(self):
+    def __insert_masked_print(self):
```

Comment:
Only seeing this on this second reading, but I would make this "semi-private", i.e., a single underscore, partly just because it seems something people might want to call, but mostly since we don't use the class mangling much elsewhere in numpy

Reply:
> we don't use the class mangling much elsewhere in numpy

We give the users great power, they have to provide the great responsibility :)

@eric-wieser I'll make the change and merge if that is OK with you.

Reply:
Go ahead.